lib/responder.js

```diff
@@ -418,27 +456,77 @@ MultipartDispatcher.prototype.objectStream = function dispatchMultipartObjectStr
   var errorListener   = operation.errorListener;
 
   var rawHeaderQueue = new FifoQueue(5);
-  var metadataBuffer = null;
+  var partReaderQueue = new FifoQueue(3);
 
   var parsingParts = 0;
   var parsedParts  = 0;
 
   var hasParsed = false;
   var hasEnded  = false;
 
-  var partTransform = function objectPartQueueTransform(data) {
-    parsedParts++;
-    var objectQueue = queuedReader.getItemQueue();
-    metadataBuffer = operation.queueDocument(
-        (data.length === 0) ? null : data, rawHeaderQueue, metadataBuffer, objectQueue
-        );
-    doneChecker();
+  var responseFinisher = function objectResponseFinisher() {
+
+    if (!partReaderQueue.hasItem() && hasParsed && hasEnded) {
+
+      // If there is metadata left in the buffer, write it
+      if (operation.nextMetadataBuffer !== null) {
+        var metadataHeaders = operation.nextMetadataBuffer[0];
+        mlutil.copyProperties(operation.nextMetadataBuffer[1], metadataHeaders);
+        operation.outputStream.write(metadataHeaders);
+        operation.nextMetadataBuffer = null;
+      }
+
+      rawHeaderQueue      = null;
+      parser              = null;
+      partHeadersListener = null;
+      partListener        = null;
+      parseFinishListener = null;
+      responseEndListener = null;
+
+      operation.outputStream.end();
+    }
+
   };
 
-  var queuedReader = new QueuedReader(
-      {objectMode: true}, operation.logger, partTransform
-      );
-  this.queuedReader = queuedReader;
+  var partFinisher = function objectPartFinisher(data) {
+    parsedParts++;
+    operation.logger.debug('parsed part %d', parsedParts);
+
+    var madeObject = operation.makeObject(
+      (data.length === 0) ? null : data, rawHeaderQueue
+    );
+
+    var writeResult = null;
+    if (madeObject !== null && madeObject !== undefined) {
+      writeResult = operation.outputStream.write(madeObject);
+    }
+
+    // Manage backpressure
+    if (writeResult === false) {
+      // Only pause stream if not all parsed
+      if (hasParsed && !partReaderQueue.hasItem()) {
```

Comment:
Maybe test whether the HTTP ReadableStream has ended?  In which case there's nothing to pause?